server/routers/authenticate/authProviders/passkeys.ts

```diff
@@ -0,0 +1,714 @@
+            publicKey = Buffer.from(verification.registrationInfo.credential.publicKey).toString('base64');
+          } else if (attestationResponse.response?.publicKey) {
+            // Try to get it from the attestation response
+            publicKey = Buffer.from(attestationResponse.response.publicKey).toString('base64');
+          } else if (attestationResponse.response?.publicKeyBytes) {
+            // Try to get it from publicKeyBytes
+            publicKey = Buffer.from(attestationResponse.response.publicKeyBytes).toString('base64');
+          } else {
+            // Use a placeholder if we can't find it
+            console.warn("Could not find public key in attestation response");
+            publicKey = "placeholder-public-key";
+          }
+        } catch (error) {
+          console.error("Error extracting public key:", error);
+          publicKey = "error-extracting-public-key";
+        }
+        
+        // Create a default label without relying on ctx.req
+        const defaultLabel = `Passkey for ${email}`;
+        
+        // Create the passkey directly with the label field
+        const passkey = await ctx.prisma.passkey.create({
+          data: {
+            userId: userProfile.supId,
+            credentialId: credentialId,
+            publicKey: publicKey,
+            signCount: verification.registrationInfo?.credential?.counter || 0,
+            createdAt: new Date(),
+            lastUsedAt: new Date(),
+            label: defaultLabel,
+          },
+        });
+        
+        console.log("Created passkey:", passkey);
+        
+        // Send magic link to the user's email for verification
+        const { error } = await supabase.auth.signInWithOtp({
+          email,
+        });
+        
+        if (error) {
+          throw new TRPCError({
+            code: "INTERNAL_SERVER_ERROR",
+            message: `Failed to send magic link: ${error.message}`,
+          });
+        }
+        
+        // Clean up challenges
+        await ctx.prisma.passkeyChallenge.deleteMany({
+          where: {
+            userId: tempUserId,
+          },
+        });
+        
+        return {
+          message: "Passkey registered successfully. Please check your email for a magic link to verify your account.",
+        };
+      } catch (error) {
+        console.error("Verification error:", error);
+        throw error;
+      }
+    }),
+
+  // Add authentication endpoints
+  startAuthentication: publicProcedure.mutation(async ({ ctx }) => {
+    try {
+      // Check if there are any passkeys in the database
+      const allPasskeys = await ctx.prisma.passkey.findMany();
+      console.log("All passkeys at start of authentication:", allPasskeys);
+      
+      // Generate authentication options
+      const options = await generateAuthenticationOptions({
+        rpID: relyingPartyID,
+        userVerification: "preferred",
+        allowCredentials: allPasskeys.map(pk => ({
```

Comment:
I think you can pass an empty array to let the browser prompt the user to select a credential.